### PR TITLE
Create changelog workflow

### PR DESCRIPTION
**Pull request by tyrion70**
This workflow checks that the CHANGELOG file was updated (or created) in each PR.

### CODE REVIEW

**Comment by kaleofduty** [comment]
Looks like the run errored?

File: .github/workflows/changelog.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  changelog:
+    name: Changelog checker
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check for changed files
+        id: changedfiles
+        uses: futuratrepadeira/changed-files@88878571ea9b265011eb41efe51c322e75e803c0
```
Comment (kaleofduty) [inline .github/workflows/changelog.yml:19]: Can you add comments with version numbers?

File: .github/workflows/changelog.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        with:
+          repo-token: ${{ github.token }}
+          pattern: '^CHANGELOG$'
+      - name: Make a comment
+        uses: unsplash/comment-on-pr@182ec73bcf2b77868dcadd16987721fcfaca76d9
```
Comment (kaleofduty) [inline .github/workflows/changelog.yml:24]: Can you add comments with version numbers?

**Comment by tyrion70** [comment]
> Looks like the run errored?

Yeah I updated the other actions to a newer version but that broke so I reverted back to the old version I was using. Also added the version numbers now in the comments

File: .github/workflows/changelog.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+name: Changelog
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the master branch
+on:
+  pull_request:
+    branches: [ develop ]
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
```
Comment (kaleofduty) [inline .github/workflows/changelog.yml:11]: I think we can get rid of this comment

File: .github/workflows/changelog.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
+# This is a basic workflow to help you get started with Actions
+
+name: Changelog
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the master branch
+on:
+  pull_request:
+    branches: [ develop ]
```
Comment (kaleofduty) [inline .github/workflows/changelog.yml:9]: Might make sense to run it on all PRs, not just into develop?

**Comment by j16r** [comment]
If I'm doing something that shouldn't go into the CHANGELOG, is there a convenient escape hatch? e.g: a chore / refactor.

File: .github/workflows/changelog.yml (hunk 4 above)
Comment (j16r) [reply .github/workflows/changelog.yml]: Might want to make sure that doesn't block releases/hotfixes etc. which don't necessarily add any new changes.

**Comment by tyrion70** [comment]
> If I'm doing something that shouldn't go into the CHANGELOG, is there a convenient escape hatch? e.g: a chore / refactor.

The PR is not blocked from merging, its just a suggestion to add information. You could just ignore that but it serves as a reminder for reviewers and the PR starter to at least consider it. Is that OK?

**Comment by kaleofduty** [comment]
> If I'm doing something that shouldn't go into the CHANGELOG, is there a convenient escape hatch? e.g: a chore / refactor.

You ignore the comment posted by the bot? It's just meant as a reminder to use your judgement whether it would make sense to add a changelog item.

**Comment by j16r** [comment]
@kaleofduty @tyrion70 yeah it wasn't clear if this was going to be blocking or suggestive, happy if it's just a comment on the PR.

**Comment by kaleofduty** [comment]
Perhaps we should edit the message to make that more clear. E.g.
"I see that you haven't updated any CHANGELOG files. Would it make sense to do so?"

**Comment by tyrion70** [comment]
That makes sense, I'll change the message to be more clear there.. @j16r do you like the format for the CHANGELOG-explorer there btw? I haven't added any historical data except the last release. Open for suggestions there.

**Comment by j16r** [comment]
Yeah it looks good, also thank you for using iso8601.

**Pull request merged**